app/src/main/java/com/github/houseorganizer/houseorganizer/RegisterEmail.java

```diff
@@ -0,0 +1,100 @@
+package com.github.houseorganizer.houseorganizer;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.auth.FirebaseAuth;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RegisterEmail extends AppCompatActivity {
+
+    private FirebaseAuth mAuth;
+    private final String TAG = "RegisterEmail";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register_email);
+
+        mAuth = FirebaseAuth.getInstance();
+
+        findViewById(R.id.reg_email_register_button).setOnClickListener(
+                v -> {
+                    if (isValidEmail() && isValidPassword()) signUpWithEmail(v);
+                }
+        );
+    }
+
+    private boolean isValidEmail() {
+        EditText email_field = findViewById(R.id.reg_enter_email);
+        String email = email_field.getText().toString();
+
+        // Regex to check valid email.
+        String regex = "^[\\w!#$%&'*+/=?`{|}~^-]+(?:\\.[\\w!#$%&'*+/=?`{|}~^-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$";
+        Pattern p = Pattern.compile(regex);
+        Matcher m = p.matcher(email);
+
+        if (m.matches()) {
+            return true;
+        } else {
+            TextView error_message = findViewById(R.id.reg_email_error_message);
+            error_message.setText(R.string.email_not_valid);
+            return false;
+        }
+    }
+
+    private boolean isValidPassword() {
+        EditText password_field = findViewById(R.id.reg_enter_password);
+        String password = password_field.getText().toString();
+        EditText password2_field = findViewById(R.id.reg_confirm_password);
+        String password2 = password2_field.getText().toString();
+
+        boolean samePasswords = password.equals(password2);
+
+        // Regex to check valid password.
+        String regex = "^(?=.*[0-9])"
+                + "(?=.*[a-z])(?=.*[A-Z])"
+                + "(?=.*[@#$%^&+=])"
+                + "(?=\\S+$).{8,20}$";
+        Pattern p = Pattern.compile(regex);
```

Comment:
The underscore character isn't in the regex list, but is used in many passwords.
Is it possible to add it ?